lib/ecl/nnc_info.cpp

```diff
@@ -151,27 +156,28 @@ void nnc_info_add_nnc(nnc_info_type * nnc_info, int lgr_nr, int global_cell_numb
   }
 }
 
+bool nnc_info_has_grid_index_list( const nnc_info_type * nnc_info, int lgr_nr ) {
+  return (nnc_info_get_vector( nnc_info , lgr_nr ) != NULL);
+}
 
-const int_vector_type * nnc_info_get_grid_index_list(const nnc_info_type * nnc_info, int lgr_nr) {
+const std::vector<int>& nnc_info_get_grid_index_list(const nnc_info_type * nnc_info, int lgr_nr) {
   nnc_vector_type * nnc_vector = nnc_info_get_vector( nnc_info , lgr_nr );
-  if (nnc_vector)
-    return nnc_vector_get_grid_index_list( nnc_vector );
-  else
-    return NULL;
+  if (!nnc_vector)
+    throw std::invalid_argument(std::string(__func__)); 
```

Comment:
This is completely new semantics; it looks like return `NULL` was completely OK previously; we can then *not* suddenly throw an exception - without really checking the callsites!

Reply:
I can only find the function in test cases outside of nnc_info.cpp.